### PR TITLE
integration changes

### PR DESCRIPTION
**Pull request by shwu-nyunai**
- adds the changes required for nyuntam <-> adapt integrations

### CODE REVIEW

**Comment by Abhrant** [comment]
@shwu-nyunai 

On runnning it, I get this error : 


08/14/2024 03-04-23 - ERROR - nyuntam.factory - [[<class 'nyuntam_adapt.factory.AdaptFactory'>]] 'NoneType' object has no attribute 'lower'
Traceback (most recent call last):
  File "/workspace/nyuntam/factory.py", line 110, in create_from_dict
    instance = caller(args)
  File "/workspace/nyuntam/nyuntam_adapt/factory.py", line 29, in __init__
    super().__init__(kwargs)
  File "/workspace/nyuntam/factory.py", line 77, in __init__
    self.__pre_init__(args)
  File "/workspace/nyuntam/nyuntam_adapt/factory.py", line 26, in __pre_init__
    super().__pre_init__(kwargs)
  File "/workspace/nyuntam/factory.py", line 83, in __pre_init__
    job_service = JobServices.get_service(
  File "/workspace/nyuntam/constants/keys.py", line 12, in get_service
    if service.lower() == "adapt":
AttributeError: 'NoneType' object has no attribute 'lower'
08/14/2024 03-04-23 - ERROR - root - Failed to create Factory instance: Factory instance could not be created.
Traceback (most recent call last):
  File "/workspace/nyuntam/main.py", line 18, in main
    factory = Factory.create_from_yaml(args.yaml_path)
  File "/workspace/nyuntam/factory.py", line 146, in create_from_yaml
    return cls.create_from_dict(args)
  File "/workspace/nyuntam/factory.py", line 115, in create_from_dict
    raise Exception("Factory instance could not be created.")
Exception: Factory instance could not be created.
08/14/2024 03-04-23 - CRITICAL - root - Critical failure in main execution: Factory instance could not be created.

**Review by Abhrant — CHANGES_REQUESTED** [review]

File: factory.py [hunk 1]
```diff
@@ -1,15 +1,30 @@
...
+    @task.setter
+    def task(self, instance: AdaptionAlgorithm) -> None:
+        self._instance = instance
+
+    def __pre_init__(self, *args, **kwargs):
+        kw = args[0]
+        task = kw.get("TASK", "text_generation")
+        subtask = kw.get("subtask", None)
+        self._algorithm_class = self.get_task(task, subtask)
+        super().__pre_init__(kwargs)
```
[the remark this answers is not in the document]
Comment (shwu-nyunai) [reply factory.py]: should be *args, **kwargs instead. fixed in the revision.

File: factory.py [hunk 2]
```diff
@@ -26,13 +41,13 @@ def __init__(self, kwargs):
         loaded_task = self.get_task(task, subtask)
         self.task = loaded_task(**asdict(adapt_params))
```
[the remark this answers is not in the document]
Comment (shwu-nyunai) [reply factory.py:43]: fixed with [AdaptionAlgorithm fix](https://github.com/nyunAI/nyuntam/pull/6/commits/68152db10b731d1f17f74acf62f79ce2dbd69993)

**Review by Abhrant — APPROVED** [review]

**Comment by Abhrant** [comment]
@Arnav0400 , we can merge this one too

**Pull request merged**